### PR TITLE
Fix misalignment of zero change in datablock with visibility:hidden. (2992)

### PR DESCRIPTION
**Pull request by johnPhillips**
## Summary

Addresses issue #2992

## Relevant technical choices

Previously the `googlesitekit-data-block__change` div in the `DataBlock` component was not rendered if there was zero change. This caused a misalignment with neighbouring blocks.
I've made the component render the div in all cases, so a zero percent change will now output `0%`, however, in this case the div is hidden with `visibility: hidden`. This hides the div without losing the space it occupies, fixing the misalignment.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
@johnPhillips, overall this PR looks good to me, but there are issues with VRT tests. Those visual changes are expected, so we will need to approve broken VR tests. Could you, please, check whether you can make the sparkline with zero changes to be on the same baseline with other charts?

google-site-kit_Dashboard_Search_Funnel_0_document_1_medium

**Comment by johnPhillips** [comment]
Hey @eugene-manuilov, thanks for the review. 

It seems like the default behaviour of the graph is to plot the line halfway up the container. Presumably because that represents zero change and change can be negative or positive. So a negative change would be plotted below the line and a positive change above. At least, that's my interpretation of what I am seeing, I might be incorrect.

Screenshot 2021-04-12 at 11 54 05

I think if that's the case then I'd have to do something funky with negative margins or `position: absolute` to move the container down by 50% to make it appear that the line is at the base of the container. 

I'm happy to give that a whirl, but thought I'd double check first. What do you think?

**Comment by eugene-manuilov** [comment]
> It seems like the default behaviour of the graph is to plot the line halfway up the container. Presumably because that represents zero change and change can be negative or positive. So a negative change would be plotted below the line and a positive change above.

We don't have negative values in the data arrays that we pass to the `Sparkline` component. That graph displays actual values, not changes from day-to-day.

> I think if that's the case then I'd have to do something funky with negative margins or position: absolute to move the container down by 50% to make it appear that the line is at the base of the container.

I think we can add the following to the Sparkline component to make it display horizontal axis at the bottom when there are all values equal to zero:

```diff
diff --git a/assets/js/components/Sparkline.js b/assets/js/components/Sparkline.js
index b1d2bde52..43f74182d 100644
--- a/assets/js/components/Sparkline.js
+++ b/assets/js/components/Sparkline.js
@@ -66,6 +66,11 @@ function Sparkline( {
                ],
        };
 
+       if ( ! data.slice( 1 ).some( ( row ) => row[ 1 ] > 0 ) ) {
+               chartOptions.vAxis.minValue = 0;
+               chartOptions.vAxis.maxValue = 1;
+       }
+
        return (
                <div className="googlesitekit-analytics-sparkline-chart-wrap">
                        <GoogleChartV2
```

File: assets/js/components/Sparkline.js [hunk 1]
```diff
@@ -66,6 +66,11 @@ function Sparkline( {
 		],
 	};
 
+	if ( ! data.slice( 1 ).some( ( row ) => row[ 1 ] > 0 ) ) {
+		chartOptions.vAxis.minValue = 0;
+		chartOptions.vAxis.maxValue = 1;
+	}
```
Comment (eugene-manuilov) [inline assets/js/components/Sparkline.js:72]: @johnPhillips could you please add a comment to explain why we need this?

Comment (johnPhillips) [reply assets/js/components/Sparkline.js:74]: Of course, sorry 👍

**Review by eugene-manuilov — APPROVED** [review]
Awesome! Thanks, @johnPhillips.

**Pull request merged**